src/hooks/useTokenDisplayData.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+
+import { type EdgeCurrencyConfig, type EdgeCurrencyWallet } from 'edge-core-js'
+import { type EdgePluginMap } from 'edge-core-js/types'
+
+import { useSelector } from '../types/reactRedux'
+import { getAllTokens } from '../util/CurrencyInfoHelpers'
+import { fixFiatCurrencyCode, getDenomFromIsoCode, getYesterdayDateRoundDownHour, zeroString } from '../util/utils'
+
+/**
+ * Returns data from tokens relevant for display
+ * */
+
+export const useTokenDisplayData = (props: { currencyConfigMap: EdgePluginMap<EdgeCurrencyConfig>, tokenId?: string, wallet: EdgeCurrencyWallet }) => {
+  const { currencyConfigMap, tokenId, wallet } = props
+  const { currencyInfo } = wallet
+  const nativeCurrencyCode = currencyInfo.currencyCode
+  const tokens = getAllTokens(currencyConfigMap[currencyInfo.pluginId])
+
+  // Get currencyCode and deomination from token
+  let currencyCode, denomination
+  if (tokenId == null || nativeCurrencyCode === tokens[tokenId].currencyCode) {
+    currencyCode = nativeCurrencyCode
+    denomination = currencyInfo.denominations[0]
+  } else {
+    const token = tokens[tokenId]
+    currencyCode = token.currencyCode
+    denomination = token.denominations[0]
+  }
```

Comment:
This can be simplified to:

```js
const { currencyCode, denominations } = tokenId == null ? currencyInfo : allTokens[tokenId]
const [denomination] = denominations
```

We don't need to worry about tokens whose `currencyCode` matches the parent chain. If we have a tokenId, it's a token - period.